symbology/render-multilayer-symbols/src/main/java/com/esri/samples/render_multilayer_symbols/RenderMultilayerSymbolsSample.java

```diff
@@ -0,0 +1,460 @@
+      addPolygonGraphicsWithMarkerSymbols(List.of(90.0), 2 * offset); // hatched vertical lines
+
+      // define vector element for a hexagon which will be used as the basis of a complex point
+      var complexPointGeometry = Geometry.fromJson("{\"rings\":" +
+        "[[[-2.89,5.0],[2.89,5.0],[5.77,0.0],[2.89,-5.0],[-2.89,-5.0],[-5.77,0.0],[-2.89,5.0]]]}");
+
+      // create the more complex multilayer graphics: a point, polygon, and polyline
+      addComplexPoint(complexPointGeometry);
+      addComplexPolygon();
+      addComplexPolyline();
+
+      // add the map view to the stack pane
+      stackPane.getChildren().addAll(mapView);
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Creates the label graphics to be displayed above each category of symbol,
+   * and adds them to the graphics overlay.
+   */
+  private void addTextGraphics() {
+    graphicsOverlay.getGraphics().addAll(
+      List.of(
+        new Graphic(new Point(-150, 50, SpatialReferences.getWgs84()),
+          getTextSymbol("MultilayerPoint\nSimple Markers")),
+        new Graphic(new Point(-80, 50, SpatialReferences.getWgs84()),
+          getTextSymbol("MultilayerPoint\nPicture Markers")),
+        new Graphic(new Point(0, 50, SpatialReferences.getWgs84()),
+          getTextSymbol("Multilayer\nPolyline")),
+        new Graphic(new Point(65, 50, SpatialReferences.getWgs84()),
+          getTextSymbol("Multilayer\nPolygon")),
+        new Graphic(new Point(130, 50, SpatialReferences.getWgs84()),
+          getTextSymbol("Multilayer\nComplex Symbols"))));
+  }
+
+  /**
+   * Creates a text symbol with the provided text and consistent styling.
+   *
+   * @param text text to be displayed by the text symbol
+   * @return the constructed text symbol
+   */
+  private TextSymbol getTextSymbol(String text) {
+    var textSymbol = new TextSymbol(
+      20,
+      text,
+      ColorUtil.colorToArgb(Color.BLACK),
+      TextSymbol.HorizontalAlignment.CENTER,
+      TextSymbol.VerticalAlignment.MIDDLE
+    );
+
+    // give the text symbol a white background
+    textSymbol.setBackgroundColor(ColorUtil.colorToArgb(Color.WHITE));
+    return textSymbol;
+  }
+
+  /**
+   * Loads a picture marker symbol layer and after it has loaded, creates a new multilayer point symbol from it.
+   * A graphic is created from the multilayer point symbol and added to the graphics overlay.
+   * 
+   * @param pictureMarkerSymbolLayer the picture marker symbol layer to be loaded
+   * @param offset the value used to keep a consistent distance between symbols in the same column
+   * 
+   */
+  private void addGraphicFromPictureMarkerSymbolLayer(PictureMarkerSymbolLayer pictureMarkerSymbolLayer, double offset) {
+    // wait for the picture marker symbol layer to load and check it has loaded
+    pictureMarkerSymbolLayer.addDoneLoadingListener(() -> {
```

Comment:
Yeah found this one a bit tricky:
- Not really a fan of adding `async` onto the end of the method name because, as you say, that creates an expectation that it will return a future, which it doesn't.
- moving the listeners into the `start` method is doable, but adds a bit of boilerplateish repetition which I've been trying to avoid so far - the sample had a lot of that before refactoring.

Could go either way at this point tbh.

Reply:
Gone the second route here, changes weren't as repetitive as I expected.